### DISCUSSION TITLE
ArrayItems.Remove 如何自定义

### DISCUSSION THREAD

**Discussion by xiaobeiOfficial** [alibaba/formily#1942] [Q&A 2.x] [answered]
1、ArrayItems.Remove 如何加二次确认？ 或者说有没有自定义的方法或者入口api
![WX20210803-185852](https://user-images.githubusercontent.com/28946184/128004601-68f8be2d-ed1c-4b78-b801-23decc87b286.png)

**Comment by janryWang** [comment, marked as answer]
直接自定义一个ConfirmRemove组件，x-component指定就行了

**Comment by xiaobeiOfficial** [comment]
`import React, { useRef } from 'react';
import { I18N } from '@common/I18N';
import { Popconfirm } from 'antd';
import InonFront from '@/components/IconFont';
import { ArrayItems } from '@formily/antd';
import styles from './index.less';

const ArrayRemoveBtn = (props) => {
  const removeRef = useRef(null);
  // const array = ArrayItems.useArray();
  const index = ArrayItems.useIndex();
  console.log(7879, props, index);
  return (
    <div className={styles['xman-array-remove-btn']}>
      <Popconfirm
        title="确认删除吗？"
        onConfirm={() => {
          removeRef.current.click();
        }}
      >
        <InonFront type="iconshanchu1" />
      </Popconfirm>
      <ArrayItems.Remove {...props}  index={index}  ref={removeRef} />
    </div>
  );
};

export default ArrayRemoveBtn;`

以上是我在问问题之前，已经自定义了一个ArrayRemoveBtn组件，我在组件中需要点击icon之后弹出Popconfirm，在onConfirm的时候再去删除（代码中的使用ref 的方法不行，可以删除，但是不是我指定的index），所以问下我怎么才能在onConfirm的时候调用某个方法来实现二次确认的删除?

**Reply by janryWang** [reply]
你都用到useArray了呀，调用array.remove不就行了么

**Reply by xiaobeiOfficial** [reply]
OK，好吧，最终解决了。。。谢谢了，不过 useArray 这个方法为啥在文档里面没有呢。。扒了源码才看到-。-
解决问题的链接附上：https://codesandbox.io/s/spring-feather-6qcg1?file=/App.tsx

**Reply by janryWang** [reply]
文档漏了。。。要不来个PR完善一下？
